kernel/mod_typing.ml

```diff
@@ -230,7 +233,10 @@ let rec check_with_mod (cst, ustate) env struc (idl,new_mp) mp reso =
 let check_with ustate env mp (sign,reso,cst) = function
   | WithDef(idl, (c, ctx)) ->
     let struc = destr_nofunctor mp sign in
-    let struc', cst = check_with_def (cst, ustate) env struc (idl, (c, ctx)) mp reso in
+    let univs = match ctx with None -> Monomorphic | Some uctx -> Polymorphic uctx in
```

Comment:
Maybe WithDef should be using this type instead of abstractcontext option?

Reply:
I tried to do that but it's a can of worms in the upper layers, so this shouldn't be done.